ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +50,594 @@ def test001_test_plan_delete_testunit(self):
+    
+        # Get the completed testplans and choose a random one
+        completed_test_plans = self.test_plan_api.get_completed_testplans(limit=500)
+        first_testplan_data = random.choice(completed_test_plans)
+        first_testunit_data_in_first_testplan = (self.test_plan_api.get_testunits_in_testplan(first_testplan_data['id']))[0]
+        testunit_number = first_testunit_data_in_first_testplan['number']
+        self.base_selenium.LOGGER.info("Choosing to change testunit with name: {}, category: {} and iterations = {}".format(
+            first_testunit_data_in_first_testplan['name'], first_testunit_data_in_first_testplan['category'], first_testunit_data_in_first_testplan['iterations']))
+    
+        # go to testunits active table and search for this testunit
+        self.test_unit_page.get_test_units_page()
+        self.base_selenium.LOGGER.info('Navigating to testunit {} edit page'.format(first_testunit_data_in_first_testplan['name']))
+        self.test_unit_page.open_filter_menu()
+        self.test_unit_page.filter_by(filter_element='test_unit:no', filter_text=testunit_number, field_type='text')
+        self.test_unit_page.filter_apply()
+
+        testunit = self.test_unit_page.result_table()[0]
+        self.test_unit_page.open_edit_page(row=testunit)
+        
+        # update the iteration and category
+        article = 'All' if first_testplan_data['article'][0] == 'all' else first_testplan_data['article'][0]
+        new_iteration = str(int(first_testunit_data_in_first_testplan['iterations']) + 1)
+        self.test_unit_page.update_testunit(number=None, name=None, category='', iterations=new_iteration, random=False)
+        new_category = self.test_unit_page.get_category().lower()
+
+        # press save and complete to create a new version
+        self.test_unit_page.save_and_create_new_version()
+
+        # go back to testplans active table
+        self.test_plan.get_test_plans_page()
+
+        # create new testplan with this testunit after creating the new version
+        second_testplan_name = self.test_plan.create_new_test_plan(material_type=first_testplan_data['materialType'],
+                        article=article, test_unit=str(testunit_number))
+
+        self.base_selenium.LOGGER.info('New testplan is created successfully with name: {}, article name: {} and material type: {}'.format(second_testplan_name, 
+                        article, first_testplan_data['materialType']))
+        
+        second_testplan_data = (self.test_plan_api.get_testplan_with_quicksearch(second_testplan_name))[0]
+        
+        first_testunit_data_in_second_testplan = (self.test_plan_api.get_testunits_in_testplan(second_testplan_data['id']))[0]
+
+        # return the database as it was for the upcoming test cases
+        self.test_plan.get_and_update_testunits_dropdown_field(cursor, db, old_testunits_searchable_from_testplans)
+        self.base_page.close_connection_with_database(db)
```

Comment:
What if for any reason your code fails/raises error before these lines?  It is gonna affect all the other test cases. Changing the backend is really dangers solution

Reply:
@0xIslamTaha This was the solution to not creating a new test plan in this testcase, as the api call isn't implemented yet.

Reply:
@SalmaaRagab I suggest keeping this test case in a branch till we get the API